server/routerlicious/packages/lambdas/src/nexus/index.ts

```diff
@@ -1097,20 +1118,38 @@ export function configureWebSocketServices(
 						socket.emit("nack", "", [nackMessage]);
 						return;
 					}
-					contentBatches.forEach((contentBatch) => {
-						const contents = Array.isArray(contentBatch)
-							? contentBatch
-							: [contentBatch];
 
-						for (const content of contents) {
+					if (supportedFeaturesMap.get(clientId)?.[feature_submit_signals_v2]) {
+						for (const signal of contentBatches as ISentSignalMessage[]) {
 							const signalMessage: ISignalMessage = {
 								clientId,
-								content,
+								content: signal.content,
+								targetClientId: signal.targetClientId,
 							};
```

Comment:
We don't really care about preserving original object and `ISentSignalMessage` is really just `ISignalMessageBase`. So why not just add `clientId` to the existing object and also preserve whatever of the other properties that were sent?